community/samples/serving/helloworld-java-quarkus/README.md

```diff
@@ -0,0 +1,419 @@
+       </profile>
+     </profiles>
+   </project>
+   ```
+
+1. Run the application locally:
+
+   ```shell
+   ./mvnw compile quarkus:dev
+   ```
+
+   Go to `http://localhost:8080/` to see your `Hello World!` message.
+
+1. In your project directory, create a file named `Dockerfile` and copy the code
+   block below into it.
+
+   ```docker
+   FROM quay.io/rhdevelopers/quarkus-java-builder as builder
+   COPY . /project
+   WORKDIR /project
+   # uncomment this to set the MAVEN_MIRROR_URL of your choice, to make faster builds
+   # ARG MAVEN_MIRROR_URL=<your-maven-mirror-url>
+   # e.g.
+   #ARG MAVEN_MIRROR_URL=http://192.168.64.1:8081/nexus/content/groups/public
+
+   RUN /usr/local/bin/entrypoint-run.sh mvn -DskipTests clean package
+
+   FROM fabric8/java-jboss-openjdk8-jdk:1.5.4
+   USER jboss
+   ENV JAVA_APP_DIR=/deployments
+
+   COPY --from=builder /project/target/lib/* /deployments/lib/
+   COPY --from=builder /project/target/*-runner.jar /deployments/app.jar
+
+   ENTRYPOINT [ "/deployments/run-java.sh" ]
+   ```
+
+1. Create a new file, `service.yaml` and copy the following service definition
+into the file. Make sure to replace `{username}` with your Docker Hub
+username.
+
+   ```yaml
+   apiVersion: serving.knative.dev/v1alpha1
+   kind: Service
+   metadata:
+     name: helloworld-java-quarkus
+   spec:
+     runLatest:
+       configuration:
+         revisionTemplate:
+           spec:
+             container:
+               image: docker.io/{username}/helloworld-java-quarkus
+               env:
+                 - name: MESSAGE_PREFIX
+                   value: "Namaste"
+   ```
+
+## Building and deploying the sample
+
+Once you have recreated the sample code files (or used the files in the sample
+folder) you're ready to build and deploy the sample app.
+
+We will be using hub.docker.com as the container registry, if you dont have an account yet have one created before proceeding further.
+
+### Using Docker
+
+Use Docker to build the sample code into a container. To build and push with
+Docker Hub, run these commands replacing `{username}` with your Docker Hub
+username:
+
+```shell
+
+# We will use the docker daemon of the minikube
+eval $(minikube docker-env)
+
+# build the jar and libs
+./mvnw clean package
+
+# Build the container on your local machine
+docker build -t {username}/helloworld-java-quarkus .
+
+# Push the container to docker registry
+docker push {username}/helloworld-java-quarkus
+```
+
+After the build has completed and the container is pushed to docker hub, you
+can deploy the app into your cluster. Ensure that the container image value
+in `service.yaml` matches the container you built in the previous step. Apply
+the configuration using `kubectl`:
+
+```shell
+kubectl apply --filename service.yaml
+```
+
+### Using Knative Build
+
+The sources also has a knative build file called `quarkus-build.yaml`, that will perform docker less container image build using [buildah](https://buildah.io)
```

Comment:
```suggestion
To use Knative Build, you add your Docker Hub information to the `quarkus-build.yaml` file.
```

Moved optional info about build to the Before you begin section.

Reply:
There is no mention of installing Knative Build until now, which is late in the flow and forces the user to go back and install the Build component if in fact thats what they prefer.